onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -713,7 +713,13 @@ class PlannerImpl {
               }
             }
           }
-        } else if (IsNonTensor(*node_output) || external_outputs) {
+        } else if (external_outputs) {
+          ORT_ENFORCE(!IsNonTensor(*node_output), "Only tensors are supported for external outputs for now.");
+          AllocPlan(current).alloc_kind = AllocKind::kPreExisting;
+#if !defined(ORT_MINIMAL_BUILD) && defined(ORT_MEMORY_PROFILE)
+          AllocPlan(current).life_interval.second = execution_plan.size();
```

Comment:
Since we are not allocating this tensor I wonder if it makes sense to make it part of the trace?

Reply:
I suppose we should treat it the same as input tensors ...

Reply:
For graph inputs (line 625-640), they are also kPreExisting and they also do this.

Reply:
great